pkg/frontend/middleware/validate.go

```diff
@@ -48,50 +48,50 @@ func Validate(env env.Interface, apis map[string]*api.Version) func(http.Handler
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceGroupNotFound, "", "Resource group '%s' could not be found.", vars["resourceGroupName"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceProviderNamespace"]; found {
 				if vars["resourceProviderNamespace"] != strings.ToLower(resourceProviderNamespace) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceNamespace, "", "The resource namespace '%s' is invalid.", vars["resourceProviderNamespace"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceNamespace, "", "The resource namespace '%s' is invalid.", vars["resourceProviderNamespace"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceType"]; found {
 				if vars["resourceType"] != strings.ToLower(resourceType) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], vars["api-version"])
 					return
 				}
 			}
 
 			if _, found := vars["resourceName"]; found {
 				if !rxResourceGroupName.MatchString(vars["resourceName"]) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
 					return
 				}
 			}
 
 			if _, found := vars["location"]; found {
 				if !strings.EqualFold(vars["location"], env.Location()) {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidLocation, "", "The provided location '%s' is malformed or invalid.", vars["location"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidLocation, "", "The provided location '%s' is malformed or invalid.", vars["location"])
 					return
 				}
 			}
 
 			if _, found := vars["operationId"]; found {
 				_, err := uuid.FromString(vars["operationId"])
 				if err != nil {
-					api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
+					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
```

Comment:
this should be a 404 if they are asking for the operation

Reply:
this is if the operationId passed is not a UUID